data-products/example_prison_data_product/application/transform.py

```diff
@@ -0,0 +1,104 @@
+import pandas as pd
+import logging
+import boto3
+import yaml
+import os
+from pathlib import Path
+
+
+logging.basicConfig()
+s3_client = boto3.client("s3")
+
+
+def get_data(bucket: str, key: str) -> pd.DataFrame:
+    response = s3_client.get_object(Bucket=bucket, Key=key)
+
+    status = response.get("ResponseMetadata", {}).get("HTTPStatusCode")
+
+    if status == 200:
+        logging.info("Data has been collected from {}/{}".format(bucket, key))
+        return pd.read_csv(response.get("Body"))
+    else:
+        logging.error(
+            "Bucket {} or file {} does not exist".format(bucket, key))
+        raise
+
+
+def get_tables(bucket, key, source_data):
+    yaml_key = os.path.join(
```

Comment:
We're tightly coupling this with the yml files. Happy to go ahead if the tight coupling of the code and config is accepted and agreed upon.

1. Are we all happy with having the yml file drive the table structure?
2. Are we going to keep the yml file for configuring.

My main concern here is tight coupling of this bit of the code and the config files. 

We could have a service that manages all of these configs, and then all we need to do is just making an API call to this service and not worry about tying this code up with config files - creating a service that manages the configs is an epic in itself.

So, I guess we can go ahead with this for now. I've answered by own question, just something to be aware of I guess.

Reply:
Currently we are going with approach of passing on the schema using yaml. But this can be improved in future based on user feedback . We might  even no use this yaml approach and adapt something else